lib/app/ExtensionConfigWatcher.js

```diff
@@ -25,34 +29,59 @@ class ExtensionConfigWatcher extends EventEmitter {
     this.chokidar = chokidar
   }
 
-  start () {
+  start (source = 'backend') {
     return new Promise((resolve) => {
-      const configPath = path.join(this.watchFolder, '*', 'extension-config.json')
-      this.watcher = this.chokidar.watch(configPath, this.options)
-        .on('all', (event, configPath) => {
-          fsEx.readJson(configPath, {throws: false}, (err, config) => {
-            if (err || !config) return
+      const check = source === 'backend' ? 'frontend' : 'backend'
+      const doStart = () => {
+        logger.debug(`Starting config watcher`)
+        const configPath = path.join(this.watchFolder, '*', 'extension-config.json')
+        this.watcher = this.chokidar.watch(configPath, this.options)
+          .on('all', (event, configPath) => {
+            fsEx.readJson(configPath, { throws: false }, (err, config) => {
+              if (err || !config) return
 
-            const parts = configPath.split(path.sep)
-            const extPath = parts.slice(0, parts.length - 1).join(path.sep)
+              const parts = configPath.split(path.sep)
+              const extPath = parts.slice(0, parts.length - 1).join(path.sep)
 
-            this.emit('configChange', {file: config, path: extPath})
+              this.emit('configChange', { file: config, path: extPath })
+            })
           })
-        })
-        .on('ready', () => resolve())
+          .on('ready', () => resolve())
+
+        if (this.options.ignoreInitial) {
+          this.watcher.removeAllListeners('ready')
+          return resolve()
+        }
+      }
 
-      if (this.options.ignoreInitial) {
-        this.watcher.removeAllListeners('ready')
-        return resolve()
+      const recheck = () => {
+        utils.previousProcess(check, this.appSettings.settingsFolder).then(otherPid => {
+          if (!otherPid) {
+            clearInterval(recheckInterval)
+            doStart()
+          }
+        })
       }
+
+      utils.previousProcess(check, this.appSettings.settingsFolder).then(otherPid => {
```

Comment:
we should call this function sth. like: checkProcess or checkForProcess, since that's what we're doing here.